src/coreclr/nativeaot/System.Private.CoreLib/src/System/CrashInfo.cs

```diff
@@ -0,0 +1,402 @@
+                RhFailFastReason.InternalError => CrashReason.InternalFailFast,
+                RhFailFastReason.UnhandledException or
+                RhFailFastReason.UnhandledExceptionFromPInvoke => CrashReason.UnhandledException,
+                _ => CrashReason.Unknown,
+            };
+
+            if (!WriteValue("reason"u8, (int)crashReason))
+                return false;
+
+            if (!WriteHexValue("thread"u8, crashingThreadId))
+                return false;
+
+            if (message != null)
+            {
+                if (!WriteValue("message"u8, message, max: 1024))
+                    return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Adds the exception info to the JSON buffer under the "exception" key. If the exception can not fit in
+        /// the triage buffer the allocations are backed out.
+        /// </summary>
+        /// <param name="key">the UTF8 name of the block</param>
+        /// <param name="exception">exception build triage block from</param>
+        /// <param name="maxMessageSize">limits the size of the exception message strings</param>
+        /// <param name="maxNumberStackFrames">limits the number of stack frames written to the triage buffer</param>
+        /// <param name="maxMethodNameSize">limits the size of the stack frame method name strings</param>
+        /// <returns>true - success, false - out of triage buffer space</returns>
+        private bool WriteExceptionWithFallback(ReadOnlySpan<byte> key, Exception exception, int maxMessageSize, int maxNumberStackFrames, int maxMethodNameSize)
+        {
+            int savedIndex = _currentBufferIndex;
+            if (!WriteExceptionHelper(key, exception, maxMessageSize, maxNumberStackFrames, maxMethodNameSize))
+            {
+                _currentBufferIndex = savedIndex;
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Adds the exception info to the JSON buffer
+        /// </summary>
+        /// <param name="key">the UTF8 name of the block</param>
+        /// <param name="exception">exception build triage block from</param>
+        /// <param name="maxMessageSize">limits the size of the exception message strings</param>
+        /// <param name="maxNumberStackFrames">limits the number of stack frames written to the triage buffer</param>
+        /// <param name="maxMethodNameSize">limits the size of the stack frame method name strings</param>
+        /// <returns>true - success, false - out of triage buffer space</returns>
+        private bool WriteExceptionHelper(ReadOnlySpan<byte> key, Exception exception, int maxMessageSize, int maxNumberStackFrames, int maxMethodNameSize)
+        {
+            if (!OpenValue(key, '{'))
+                return false;
+
+            if (!WriteHexValue("address"u8, (ulong)Unsafe.AsPointer(ref exception)))
```

Comment:
What happens if the pointer size is not the same as `ulong`?

Reply:
We really have two options - always use ulong or just use `nuint`

Reply:
I'm going to always using ulong.  For 32bit pointers the cast should work just fine.